### PR TITLE
TypedObjectTypingResult keeps fields order

### PR DESCRIPTION
**Pull request by nadberezny**

### CODE REVIEW

File: engine/api/src/main/scala/pl/touk/nussknacker/engine/api/typed/typing.scala [hunk 1]
```diff
@@ -37,16 +38,21 @@ object typing {
...
+    def apply(fields: List[(String, TypingResult)], objType: TypedClass): TypedObjectTypingResult =
+      TypedObjectTypingResult(ListMap(fields: _*), objType)
 
+    def apply(fields: ListMap[String, TypingResult]): TypedObjectTypingResult =
+      TypedObjectTypingResult(fields, TypedClass(classOf[java.util.Map[_, _]], List(Typed[String], Unknown)))
   }
 
-  case class TypedObjectTypingResult(fields: Map[String, TypingResult],
-                                     objType: TypedClass, additionalInfo: Map[String, AdditionalDataValue] = Map.empty) extends SingleTypingResult {
+  case class TypedObjectTypingResult(fields: ListMap[String, TypingResult],
```
Comment (arkadius) [inline engine/api/src/main/scala/pl/touk/nussknacker/engine/api/typed/typing.scala:53]: 1. Please left the comment like:
`Warning, fields are kept in list-like map so: 1) order is important 2) lookup has O(n) complexity`
2. Please add notice in `MigrationGuide.md` about this change

**Review by arkadius — APPROVED** [review]
LGTM, but please add some documentation in code + migration guide

File: engine/api/src/main/scala/pl/touk/nussknacker/engine/api/typed/typing.scala (hunk 1 above)
Comment (nadberezny) [reply engine/api/src/main/scala/pl/touk/nussknacker/engine/api/typed/typing.scala:53]: :+1:

Comment (nadberezny) [reply engine/api/src/main/scala/pl/touk/nussknacker/engine/api/typed/typing.scala:54]: https://github.com/TouK/nussknacker/pull/1532/commits/49e9f129a8634a4ee724d7774e7b254480bd48f7

**Pull request merged**